### PR TITLE
[ui] Deployment History search

### PR DESCRIPTION
**Pull request by philrenaud**
- Adds a "loading shimmer" and rolling slide-in for events:
image

- Adds a searchbox:
image

Search term persists as a deployment rolls forward; which is to say, if you search a term when there are no events that match it, but a placed allocation matches it later, it will show up with no further user action needed.

### CODE REVIEW

File: ui/mirage/factories/allocation.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ const REF_TIME = new Date();
 export default Factory.extend({
   id: (i) => (i >= 100 ? `${UUIDS[i % 100]}-${i}` : UUIDS[i]),
 
-  jobVersion: 1,
+  jobVersion: 0,
```
Comment (philrenaud) [inline ui/mirage/factories/allocation.js:16]: TBD: jobVersion is set to 1 in our tests, but Nomad jobVersions start at 0. Setting this to 0 to try to make Mirage a little closer to real life. If this breaks a lot of things, will revert.

File: ui/mirage/factories/task-event.js [hunk 2]
```diff
@@ -12,5 +12,7 @@ export default Factory.extend({
   exitCode: () => null,
   time: () => faker.date.past(2 / 365, REF_TIME) * 1000000,
 
-  displayMessage: () => faker.lorem.sentence(),
+  // TODO: I think this is probably an archaic property name. See if tests collapse with this removal and decide to remove then.
+  // displayMessage: () => faker.lorem.sentence(),
+  message: () => faker.lorem.sentence(),
```
Comment (ChaiWithJai) [inline ui/mirage/factories/task-event.js:17]: `blocking`: I'm getting 99 test failures. ~97 of them are associated to this diff. Unsure if this is the root of all the test failures...

File: ui/app/components/job-status/deployment-history.js [hunk 3]
```diff
@@ -71,4 +72,25 @@ export default class JobStatusDeploymentHistoryComponent extends Component {
       color: 'critical',
     });
   }
+
+  // #region search
+
+  /**
+   * @type { string }
+   */
+  @tracked searchTerm = '';
```
Comment (ChaiWithJai) [inline ui/app/components/job-status/deployment-history.js:81]: `question`: I noticed that this is not associated to a query parameter, as you've requested in previous PRs. Any color about why we're not using an associated query parameter here?

**Review by ChaiWithJai — CHANGES_REQUESTED** [review]
There's many test failures associated with the Mirage change.

File: ui/mirage/factories/task-event.js (hunk 2 above)
Comment (philrenaud) [reply ui/mirage/factories/task-event.js:17]: Hi @ChaiWithJai -- please see the code comment TODO in the code here. This was run to see if tests collapse.

File: ui/app/components/job-status/deployment-history.js (hunk 3 above)
Comment (philrenaud) [reply ui/app/components/job-status/deployment-history.js:81]: Good question! In this case, this is a conditional component — it only exists when a deployment is active (a much smaller % of the time than when it's in steady state).

If I wanted to add a queryParam that controlled this from the job index route, there would be a majority of time where that queryParam did nothing.

There's an argument to be made for including it anyway, but it seems pretty outweighed by the argument that each deployment is separate and search terms aren't necessarily something you'd want to maintain between all of them.

Comment (ChaiWithJai) [reply ui/app/components/job-status/deployment-history.js:81]: Gotcha!

**Review by ChaiWithJai — APPROVED** [review]

**Pull request merged**